internal/controller/services/monitoring/monitoring_controller_support.go

```diff
@@ -170,9 +192,20 @@ func getTemplateData(ctx context.Context, rr *odhtypes.ReconciliationRequest) (m
 		if err := addTracesTemplateData(templateData, traces, monitoring.Spec.Namespace); err != nil {
 			return nil, err
 		}
+		if err := setResourceValues(monitoring, templateData, "TempoStack"); err != nil {
+			return nil, err
+		}
 	}
 
-	templateData["CollectorReplicas"] = monitoring.Spec.CollectorReplicas
+	if err := setResourceValues(monitoring, templateData, "Collector"); err != nil {
+		return nil, err
+	}
+
+	// if metrics or traces are configured, set collector replicas
+	if (monitoring.Spec.Metrics != nil && (monitoring.Spec.Metrics.Storage != nil || monitoring.Spec.Metrics.Resources != nil)) ||
+		monitoring.Spec.Traces != nil {
+		templateData["CollectorReplicas"] = monitoring.Spec.CollectorReplicas
```

Comment:
Shouldn't we run here `setResourceValues(monitoring, templateData, "Collector")` from above, so collector metrics are set only if it is actually being deployed?